.github/workflows/_calculate-docker-image.yml

```diff
@@ -0,0 +1,37 @@
+name: calculate-docker-image
+
+on:
+  workflow_call:
+    inputs:
+      docker-image-name:
+        required: true
+        type: string
+        description: Name of the base docker image to build with.
+
+    outputs:
+      docker-image:
+        value: ${{ jobs.calculate-docker-image.outputs.docker-image }}
+        description: The docker image containing the built PyTorch.
+
+jobs:
+  calculate-docker-image:
+    if: github.repository_owner == 'pytorch'
+    runs-on: [self-hosted, linux.large]
+    timeout-minutes: 15
+    outputs:
+      docker-image: ${{ steps.calculate-docker-image.outputs.docker-image }}
+    steps:
+      - name: Checkout PyTorch
+        uses: pytorch/pytorch/.github/actions/checkout-pytorch@master
+        with:
+          submodules: false
+          fetch-depth: 1
```

Comment:
This will fail for every PR that has more than one commit, as one can clearly see in https://github.com/pytorch/pytorch/actions/runs/4389777316/jobs/7687694067#step:4:68

Reply:
Yeah, a fix is coming.  I have created this issue https://github.com/pytorch/pytorch/issues/96471 few days ago to document this.

Reply:
Fix https://github.com/pytorch/pytorch/pull/96588